### ISSUE TITLE
MorphOS First person rendering bug + more

### ISSUE THREAD

**Issue by Cowcat5150** [openmoh/openmohaa#519]
As seen in video (timestamp 1:13) : Enabling the normal view all is messed up.

https://www.youtube.com/watch?v=QXG-1LtD9kY&t=7s

This is happening since the beginning of doing the MorphOS port.

- Hiding the gun (holster), all is correct for first view.
- Disabling the gun rendering in menu seems to be correct but at the time you shoot, the fire animation exposes the bug again.

A couple of more issues:

- Colors in models seem to go almost black depending on light environment, compared to a Linux version that is smoother and don´t go fully dark.
- Models seems to dissolve in certain terrain bumps: For example dead bodies almost disappear.

**Comment by smallmodel** [comment]
Could you try one of the following:
- Set `r_nocull` to 1
- If that doesn't work, remove `RF_DEPTHHACK` flag from https://github.com/openmoh/openmohaa/blob/23f776d3c79ac56c9cf60010fcd0c15fbc31c37c/code/cgame/cg_modelanim.c#L1068 and https://github.com/openmoh/openmohaa/blob/23f776d3c79ac56c9cf60010fcd0c15fbc31c37c/code/cgame/cg_modelanim.c#L1290

For light problem, try toggling `r_fastentlight` (Full entity lighting in the 3D options)

**Comment by Cowcat5150** [comment]
Yep. Removing DEPTHHACK fixes the principal issue. Thanks !

**Comment by smallmodel** [comment]
Try again with the latest commit and by re-adding RF_DEPTHHACK

**Comment by Cowcat5150** [comment]
Yes, done. As soon you fixed depthmask values I saw where the problem was. Terrain still with issues with models but game now is basically what it should be.

**Comment by Cowcat5150** [comment]
Latest terrain upgrades fixed the problems.
